cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpServiceRequest.java

```diff
@@ -43,10 +45,17 @@
 
   /**
    * @return the headers of this request; each header name can map to multiple values
+   * @deprecated Use {@link #getAllHeaders()} instead. This method will be removed in future release.
    */
+  @Deprecated
   Multimap<String, String> getHeaders();
 
   /**
+   * @return all headers of this request; each header name can map to multiple values
+   */
+  Map<String, List<String>> getAllHeaders();
```

Comment:
nit: Should we add a method description in this javadoc, other than the description of ``@return``, or do you think that is unnecessary?